bofhound/ad/adds.py

```diff
@@ -126,6 +133,24 @@ def import_objects(self, objects):
                 elif 'container, groupPolicyContainer' in object_class:
                     bhObject = BloodHoundGPO(object)
                     target_list = self.gpos
+                # grab PKIs
+                elif 'top, certificationAuthority' in object_class:
+                    if 'CN=AIA,' in object.get('distinguishedname'):
+                        bhObject = BloodHoundAIACA(object)
+                        target_list = self.aiacas
+                    elif 'CN=Certification Authorities,' in object.get('distinguishedname') :
+                        bhObject = BloodHoundRootCA(object)
+                        target_list = self.rootcas
+                elif 'top, pKIEnrollmentService' in object_class:
+                    bhObject = BloodHoundEnterpriseCA(object)
+                    target_list = self.enterprisecas
+                # grab PKI Templates
+                elif 'top, pKICertificateTemplate' in object_class:
+                    bhObject = BloodHoundCertTemplate(object)
+                    target_list = self.certtemplates
+                elif 'top, container' in object_class:
+                    bhObject = BloodHoundContainer(object)
+                    target_list = self.containers
```

Comment:
Something to test - I wonder if this was caused by querying the Configuration naming context for `objectclass=*` instead of specific ADCS related objects

Reply:
I have one guess regarding this question, can already apply a first filter. This filter may need to be updated:

`"CN=5E1574F6-55DF-493E-A671-AAEFFCA6A100,CN=OPERATIONS,CN=DOMAINUPDATES,CN=SYSTEM,DC=GALAXY,DC=LAN"` and `"distinguishedname": "CN=MACHINE,CN={6AC1786C-016F-11D2-945F-00C04FB984F9},CN=POLICIES,CN=SYSTEM,DC=GALAXY,DC=LAN",` are not collected by Sharphound

Adding a filter on `\{.*\},CN=CN=POLICIES,CN=SYSTEM,` regex and `CN=OPERATIONS,CN=DOMAINUPDATES,CN=SYSTEM` string

Maybe can we just filter on `CN=SYSTEM` ?